### PR TITLE
Minor code improvements, add documentation for nspace and target modules

### PR DESCRIPTION
**Pull request by Geod24**
Better reviewed commit-by-commit, ignoring whitespace diff.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: src/dmd/nspace.d [hunk 1]
```diff
@@ -1,7 +1,44 @@
...
+ *
+ * Note that it implies `extern(C++)` so it cannot be used as a generic
+ * named scope. Additionally, `Nspace` with the same `Identifier` can be
+ * defined in different module (as C++ allows a namespace to be spread accross
+ * translation units), but symbols in it should be considered
+ * part of the same scope. Lastly, not all possible C++ namespace names
+ * are valid D identifier.
+ *
+ * Those shortcomings led to the development of the string variant
+ * (`dmd.attrib : CPPNamespaceDeclaration`), which should be preferred.
```
Comment (WalterBright) [inline src/dmd/nspace.d]: I don't agree with the last two lines. Please stop editorializing this.

File: src/dmd/dsymbol.d [hunk 2]
```diff
@@ -1042,51 +1042,54 @@ extern (C++) class Dsymbol : ASTNode
     {
         //printf("Dsymbol::oneMembers() %d\n", members ? members.dim : 0);
         Dsymbol s = null;
-        if (members)
+        if (!members)
         {
-            for (size_t i = 0; i < members.dim; i++)
+            *ps = null;
```
Comment (WalterBright) [inline src/dmd/dsymbol.d]: Adding this line changes the behavior, making this change not trivial and not a clear refactor.

File: src/dmd/nspace.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/nspace.d]: Yes, I know your opinion on this, and you know mine.

**Review by WalterBright — CHANGES_REQUESTED** [review]
per my comment.

File: src/dmd/nspace.d (hunk 1 above)
Comment (Geod24) [reply src/dmd/nspace.d]: Well, those shortcomings *did* lead to the development of the string variant, that's not an opinion. I'll remove the "should be preferred" though.
And I still strongly think it should be deprecated. It's been ~16 months and I haven't seen a single person happy with this feature, and none of the bindings in druntime use it.

Comment (Geod24) [reply src/dmd/nspace.d]: Actually removed the whole sentence, the previous paragraph should be enough.

Comment (WalterBright) [reply src/dmd/nspace.d]: thank you

**Review by WalterBright — APPROVED** [review]

**Pull request merged**